tests/ga/test_monitor.py

```diff
@@ -276,6 +279,38 @@ def test_failed_heartbeat_creates_telemetry(self, patch_report_heartbeat, _, *ar
         self.assertEqual(False, args[5].call_args[1]['is_success'])
         monitor_handler.stop()
 
+    @patch('azurelinuxagent.common.logger.Logger.info')
+    def test_reset_loggers(self, mock_info, *args):
+        # Adding 100 different messages
+        for i in range(100):
+            event_message = "Test {0}".format(i)
+            logger.periodic_info(logger.EVERY_DAY, event_message)
+
+            self.assertIn(hash(event_message), logger.DEFAULT_LOGGER.periodic_messages)
+            self.assertEqual(i + 1, mock_info.call_count)  # range starts from 0.
+
+        self.assertEqual(100, len(logger.DEFAULT_LOGGER.periodic_messages))
+
+        # Adding 1 message 100 times, but the same message. Mock Info should be called only once.
+        for i in range(100):
+            logger.periodic_info(logger.EVERY_DAY, "Test-Message")
+
+        self.assertIn(hash("Test-Message"), logger.DEFAULT_LOGGER.periodic_messages)
+        self.assertEqual(101, mock_info.call_count)  # 100 calls from the previous section. Adding only 1.
+        self.assertEqual(101, len(logger.DEFAULT_LOGGER.periodic_messages))  # One new message in the hash map.
+
+        # Resetting the logger time states.
+        monitor_handler = get_monitor_handler()
+        monitor_handler.last_reset_loggers_time = datetime.datetime.utcnow() - timedelta(hours=1)
+        MonitorHandler.RESET_LOGGERS_PERIOD = timedelta(milliseconds=100)
```

Comment:
Not needed?

Reply:
+1, and this part of the test should be done directly on the logger, rather than on the logger via the monitor (plus move test to the logger tests)

Reply:
(could be done separately, in the interest of moving forward with the release)

Reply:
@pgombar 
It is needed. We are not using the `RESET_LOGGERS_PERIOD` for any sleep, but for the calculation to reset the logger. I am just moving the last time sent to 1 hour, but also need to force the calculation of time_now >= _now(-1hr) + 100ms_ to be true, to hit that code.

```
if time_now >= (self.last_reset_loggers_time + MonitorHandler.RESET_LOGGERS_PERIOD):
    logger.reset_periodic()
```

@narrieta: This part of the test is testing the monitor pieces of the code and not the logger pieces of the code. We already test the periodic_* in the test_loggers (eg: `test_periodic_does_not_emit_if_previously_sent`). Yes, there needs to be a test specifically for only `logger.reset_periodic()` in the test_logger, and that can be added later (will send a new PR later for that, against develop).

Reply:
yup, my bad... good as it is now. thanks!